File: game-server/src/main/java/net/dodian/uber/game/model/EntityType.java
package net.dodian.uber.game.model;

/**
 * Represents different types of entities in the game world.
 * Used for segregating entities in chunk storage.
 */
public enum EntityType {
    /**
     * Player entity type.
     */
    PLAYER,

    /**
     * NPC entity type.
     */
    NPC,

    /**
     * Static object entity type.
     */
    OBJECT,

    /**
     * Ground item entity type.
     */
    GROUND_ITEM;
}

File: game-server/src/main/kotlin/net/dodian/uber/game/modelkt/entity/EntityType.kt
package net.dodian.uber.game.modelkt.entity

enum class EntityType {
    DYNAMIC_OBJECT,
    GROUND_ITEM,
    NPC,
    PLAYER,
    PROJECTILE,
    STATIC_OBJECT
}

val EntityType.isMob get() = this == EntityType.PLAYER || this == EntityType.NPC
val EntityType.isTransient get() = this == EntityType.PROJECTILE